src/kaleidoscope/driver/keyscanner/ATMega.h

```diff
@@ -0,0 +1,242 @@
+/* -*- mode: c++ -*-
+ * ATMegaKeyScanner -- ATMega-based keyscanner component
+ * Copyright (C) 2018-2019  Keyboard.io, Inc
+ *
+ * This program is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License as published by the Free Software
+ * Foundation, version 3.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include <Arduino.h>
+#include "Kaleidoscope-HIDAdaptor-KeyboardioHID.h"
+
+#include "kaleidoscope/macro_helpers.h"
+#include "kaleidoscope/driver/keyscanner/Base.h"
+#include "kaleidoscope/hardware/avr/pins_and_ports.h"
+
+#include <avr/wdt.h>
+
+#define ROW_PIN_LIST(...)  __VA_ARGS__
+#define COL_PIN_LIST(...)  __VA_ARGS__
+
+#define ATMEGA_KEYSCANNER_PROPS(ROW_PINS_, COL_PINS_)               \
+  KEYSCANNER_PROPS(NUM_ARGS(ROW_PINS_), NUM_ARGS(COL_PINS_));       \
+  static constexpr uint8_t matrix_row_pins[matrix_rows] =  ROW_PINS_;   \
+  static constexpr uint8_t matrix_col_pins[matrix_columns] =  COL_PINS_;
+
+#define ATMEGA_KEYSCANNER_PROPS_IMPLEMENTATION(BOARD)   \
```

Comment:
I think this is a private macro. Is that right? If so, can we use a standard naming convention to make sure it stays private and not used from elsewhere? (And maybe even undef it at the end of the file)

Reply:
I'd rather keep it public, in case anyone wants to subclass `keyscanner::ATMega`. It could be useful there.

Reply:
Let's make it public the first time that comes up? It's easier to make things public later than make them private later.

Reply:
Also, should we be using _BOARD or something to make it clear that that's a local param and not a global define?